### Primer Parcial --> Matematicas Avanzadas para Ciencia de Datos

Importacion de libreria de metodo simplex codificada en Python ->

In [2]:
import simplex_program as sp 

<img src='./1.png' style="width:800px;">

- Programacion lineal maximizar z = cx donde $A_x$ $\leq$ b , z $\geq$ 0
- Toma un punto w que  $A_x$ <$b_1 x_o$> 0 
- $A_(x0)$ tenemos que para E>theta que existe x´= $x_0 + C$ dado que
- $A_x' \leq , x' > theta$
- $A_x' \leq b $ implica que:

A($x_0$ + E) $\leq$ b
A$x_0$ + AE $\leq$ b
b $\geq$ A$x_0$ + A0(theta)
b $\geq$ b + A 0 --> que es imposible

Tambien ($X_0 \leq cx´$ , esto implica que esta dentro de la region feasiblem que podemos concluir que $x_0 no es una solucion optima$ )



<img src='./2.png' style="width:800px;">

a)
<img src='./G2.jpg' style="width:300px;">

maxim 10 $\Lambda_2$ + 24 $\Lambda_3$ + 12 $\Lambda_4$
-> factible solution = 24

- $x_1 + 3x_2$ --> 1(0) + 3(0) = 0
- -$x_1$ + $x_2$ <= 4 ---> 1(0) + 3(4) = 12
- -$x_1$ + $2x_2$ <= 12 ---> 1(4) + 3(8)

Observando graficamente no se observa que exista una zona feasible posible para la maximizacion

<img src='./3.png' style="width:800px;">

<img src='./G1.jpg' style="width:300px;">

A) 
- $x_1 - x_2 < 4$
- $x_1 > 0$
---> (4,0)
B) 
- $x_1 + 2x_2 > 6$
- $x_1 + 2x_2 <2$
---> (4.67,0.67) 
C)
- $x_1 + 2x_2 < 6$
- $x_2 < 2$
---> (2,2)
D)
- $x_2 <2$
- $x_1 > 0$
---> (0,2)

<img src='./4.png' style="width:800px;">

<img src='./G3.jpg' style="width:300px;">

No se puede obtener una region feasible dado que no tiene limitantes , entonces existe una region infinita

<img src='./5.png' style="width:800px;">

<img src='./G4.jpg' style="width:300px;">

Si, esta en el area de las feasibles

<img src='./6.png' style="width:800px;">

a)
<img src='./G5.jpg' style="width:300px;">

In [3]:
#las 3 posiciones extra en el arreglo ya incluye los valores de pivoteo
c = [5, 4, 0, 0, 0]
#aqui se incluyen los valores de pivoteo
A = [
    [ 1, 2, 1, 0, 0],
    [ 2, -1, 0, 1, 0],
    [ 5, 3, 0, 0, 1]
]
#se muestran los valores para despues de la desigualdad
b = [6, 4, 15]
sp.simplex(c,A,b)

[[1, 2, 1, 0, 0, 6], [2, -1, 0, 1, 0, 4], [5, 3, 0, 0, 1, 15]]
[5, 4, 0, 0, 0, 0]


[1.714285714285714, 2.1428571428571432, 0, 2.7142857142857153, 0, 0]

La solucion optima es : $x_1=1.71,x_2=2.14$ y maximo $Z=17.14.$

<img src='./7.png' style="width:800px;">

In [6]:
#las 3 posiciones extra en el arreglo ya incluye los valores de pivoteo
c = [3, 2, 0, 0]
#aqui se incluyen los valores de pivoteo
A = [
    [ 2, -3, 1, 0],
    [ -1, 1, 0, 1]
    #[ 5, 3, 0, 0, 1]
]
#se muestran los valores para despues de la desigualdad
b = [3, 5]
#sp.simplex(c,A,b)

Este problema va a ocasionar un desbordamiento -->

<img src='./8.png' style="width:800px;">

$x_1 + x_2 + x_3 = 1$
<br>
$25x_1 + 15x_2 + 25x_3 + x_5 = 22$
<br>
$15x_1 + 30x_2 + 20x_3 - x_6 = 20$
<br>
$5x_1 + 12x_2 + 8x_3 - x_7 = 6$
<br>
$5x_1 + 12x_2 + 8x_3 + x_8 = 12$
<br>
$x_j = 0 $
<br>


\begin{equation}
\begin{bmatrix}
1 & 0 & 0 & 0 & 0 & 0\\
25 & -1 & 0 & 0 & 0 & 0\\
-25 & 0 & 0 & 0 & 0 & 0\\
20 & 0 & 0 & -1 & 0 & 0\\
8 & 0 & 0 & 0 & -1 & 0 \\
-8 & 0 & 0 & 0 & 0 & 1
\end{bmatrix}
\end{equation}

$x_B$ = 
\begin{equation}
\begin{bmatrix}
x_3 \\
x_4 \\
x_5 \\
x_6 \\
x_7 \\
x_8
\end{bmatrix}
\end{equation}

$x_3$ = 1- $x_1$ - $x_2$
<br>
$x_4$ = 7 - 10 $x_2$
z = 0.116 + 0.3 $x_3$ + 0.012 $x_4$

$x_B$ = 
\begin{equation}
\begin{bmatrix}
x_1 = 0.3 \\
x_2 = 0.7 \\
x_3 = 0 \\
x_4 = 0\\
x_5 = 4\\
x_6 = 5.5\\
x_7 = 3.9\\
x_8 = 2.1
\end{bmatrix}
\end{equation}

<img src='./9.png' style="width:800px;">

Respuesta: 
<br>
Maximizar Z = 2  $x_1$ + 4  $x_2$ + 2.5  $x_3$

Sujeto a: 
<br>
$3 x_1 + 4 x_2 + 2 x_3 < 600$
<br>
$2 x_1 + x_2 + 2 x_3 < 400$
<br>
$x_1 + 3x_2 + 3x_3 < 300$
<br>
Condicion de no negatividad
($x_1 , x_2 , x_3 > 0 $)

In [10]:
#las 3 posiciones extra en el arreglo ya incluye los valores de pivoteo
c = [2, 4, 2.5, 0, 0 , 0]
#aqui se incluyen los valores de pivoteo
A = [
    [ 3, 4, 2, 1, 0 , 0],
    [ 2, 1, 2, 0, 1 , 0],
    [ 1, 3, 3, 0, 0 , 1]
]
#se muestran los valores para despues de la desigualdad
b = [600, 400, 300]
sp.simplex(c,A,b)

[[3, 4, 2, 1, 0, 0, 600], [2, 1, 2, 0, 1, 0, 400], [1, 3, 3, 0, 0, 1, 300]]
[2, 4, 2.5, 0, 0, 0, 0]


[120.0, 60.0, 0, 0, 99.99999999999999, 0, 0]

Obtenemos que las respuestas optimas son: 
$x_1$ = 120
$x_2$ = 60
$x_3$ = 0

<img src='./10.png' style="width:800px;">

Supongamos  $x,y,z$ son el precio de los peanuts, cashews y almonds por onza
<br>
Asi que que por las condionales dadas $->$
<br>
$2.4x + 6y + 3.6z = 1.3$ -- (i)
<br>
$6x + 3.6y + 2.4z = 1.1$ -- (ii)
<br>
$10.8 x + 1.2 y = 0.8$ -- (iii)
<br>
Pasemos la matriz a python -> 😗


In [16]:
#las 3 posiciones extra en el arreglo ya incluye los valores de pivoteo
c = [1, 1, 1, 0, 0 , 0]
#aqui se incluyen los valores de pivoteo
A = [
    [ 2.4, 6, 3.6, 1, 0 , 0],
    [ 6, 3.6, 2.4, 0, 1 , 0],
    [ 10.8, 1.2, 0, 0, 0 , 1]
]
#se muestran los valores para despues de la desigualdad
b = [1.3, 1.1, 0.8]
sp.simplex(c,A,b)

[[2.4, 6, 3.6, 1, 0, 0, 1.3], [6, 3.6, 2.4, 0, 1, 0, 1.1], [10.8, 1.2, 0, 0, 0, 1, 0.8]]
[1, 1, 1, 0, 0, 0, 0]


[0.05303030303030308, 0, 0.3257575757575757, 0, 0, 0.2272727272727266, 0]

Peanut = 0.05 * (150) = 30
<br>
cashew = 0(100) = 0
<br>
almonds = 0.3(50) = 15

Entonces el total = 30x0.06+50*0.1041 + 15*0.14 = 9.270845
N = 9.27 / 12 + 0.00625
N = 12.26 *Cans*

<img src='./11.png' style="width:800px;">

En la tercera iteracion es cuando se desborda el sistema ->

<img src='./12.png' style="width:800px;">

<img src='./13.png' style="width:800px;">

In [ ]:
#las 3 posiciones extra en el arreglo ya incluye los valores de pivoteo
c = [1, 1, 0, 0, 0]
#aqui se incluyen los valores de pivoteo
A = [
    [-1, 1, 1, 0, 0],
    [ 1, 0, 0, 1, 0],
    [ 0, 1, 0, 0, 1]
]
#se muestran los valores para despues de la desigualdad
b = [2, 4, 4]
sp.simplex(c,A,b)